src/ng/filter/orderBy.js

```diff
@@ -460,6 +460,9 @@
  * way. (When specifying a custom comparator, you also need to pass a value for the `reverse`
  * argument - passing `false` retains the default sorting order, i.e. ascending.)
  *
+ * If your comparator fails to differentiate between two items, they will fall back to the built-in
+ * comparator function, this produces a guarantee of stable sorting.
+ *
```

Comment:
Actually, this is the wrong place for this comment. It should be moved to [L55][1] and be changed to something like:

> If a custom comparator cannot still can't distinguish between two items, then they will be sorted based on their index using the built-in comparator.

[1]: https://github.com/BobChao87/angular.js/blob/8f236af514d749f194c0a1ef8fa7f692386997eb/src/ng/filter/orderBy.js#L55